### PR TITLE
feat(task): fluent expression WIP

### PR DESCRIPTION
**Pull request by Guikingone**

### CODE REVIEW

File: doc/tasks.md [hunk 1]
```diff
@@ -2,20 +2,21 @@
...
 
 ## Extra
 
-- [Task callbacks](tasks.md#Callbacks)
-- [Task notifications](tasks.md#Notifications)
-- [Options](tasks.md#Options)
+- [Task callbacks](#Callbacks)
+- [Task notifications](#Notifications)
+- [Options](#Options)
+- [Fluent Expression](#Expression)
```
Comment (jmsche) [inline doc/tasks.md:19]:
```suggestion
- [Fluent Expressions](#fluent-expressions)
```

File: doc/tasks.md [hunk 2]
```diff
@@ -180,3 +181,14 @@ Each task can define a set of notification:
 ## Options
 
 Each task has its own set of options, the full list is documented in [AbstractTask](../src/Task/AbstractTask.php).
+
+## Expression
+
+This bundle support defining tasks expression via basic cron syntax:
```
Comment (jmsche) [inline doc/tasks.md:187]:
```suggestion
This bundle supports defining tasks expression via basic cron syntax:
```

File: doc/tasks.md [hunk 3]
```diff
@@ -180,3 +181,14 @@ Each task can define a set of notification:
 ## Options
 
 Each task has its own set of options, the full list is documented in [AbstractTask](../src/Task/AbstractTask.php).
+
+## Expression
```
Comment (jmsche) [inline doc/tasks.md:185]:
```suggestion
## Fluent Expressions
```

File: doc/tasks.md [hunk 4]
```diff
@@ -180,3 +181,14 @@ Each task can define a set of notification:
...
+
+This bundle support defining tasks expression via basic cron syntax:
+
+```bash
+* * * * *
+```
+
+Even if this approach is mostly recommended, you may need to use a more "user-friendly" syntax, to do so,
+this bundle allows you to use "fluent" expressions thanks to [strtotime](https://www.php.net/manual/fr/function.strtotime).
```
Comment (jmsche) [inline doc/tasks.md:194]: Maybe add an example here?

**Review by jmsche — CHANGES_REQUESTED** [review]

File: doc/tasks.md (hunk 2 above)
Comment (Guikingone) [reply doc/tasks.md:187]: Fixed

File: doc/best_practices.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Best practices
+
+- [Transports](#Transports)
+- [External ressources](#Others)
```
Comment (jmsche) [inline doc/best_practices.md:4]:
```suggestion
- [External ressources](#external-resources)
```

File: doc/best_practices.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+## Transports
+
+When using transports that rely on network calls, you must keep in mind that this transport
```
Comment (jmsche) [inline doc/best_practices.md:8]:
```suggestion
When using transports that rely on network calls, you must keep in mind that these transports
```

File: doc/best_practices.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+can fail to retrieve/create and so on, it could be a good idea to use the fail over or round robin
+transport to prevent any errors:
+
+```yaml
+scheduler_bundle:
+    transport:
+        dsn: 'failover://(doctrine://... || fs://last_in_first_out)'
+```
+
+## Others
```
Comment (jmsche) [inline doc/best_practices.md:18]:
```suggestion
## External resources
```

**Review by jmsche — CHANGES_REQUESTED** [review]

File: doc/best_practices.md (hunk 7 above)
Comment (Guikingone) [reply doc/best_practices.md:18]: Fixed

File: doc/tasks.md (hunk 3 above)
Comment (Guikingone) [reply doc/tasks.md:185]: Fixed

File: doc/best_practices.md (hunk 6 above)
Comment (Guikingone) [reply doc/best_practices.md:8]: Fixed

File: doc/best_practices.md (hunk 5 above)
Comment (Guikingone) [reply doc/best_practices.md]: Fixed

File: doc/tasks.md (hunk 1 above)
Comment (Guikingone) [reply doc/tasks.md]: Fixed

File: doc/tasks.md (hunk 4 above)
Comment (Guikingone) [reply doc/tasks.md]: Improved

**Pull request merged**